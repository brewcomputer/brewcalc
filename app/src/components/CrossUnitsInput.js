import React from 'react'
import { connect } from 'react-redux'

const CrossUnitsInput = ({ name, description, value, unit, units }) => {
  return (
    <li title={description}><b>{name}: </b>{value} {unit}</li>
  )
}

const mapStateToProps = ({ units }) => ({ units })

export default connect(mapStateToProps)(CrossUnitsInput)
