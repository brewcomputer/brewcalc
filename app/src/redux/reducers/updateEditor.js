import { recipe } from '../../data/recipe'
import { equipment } from '../../data/equipment'

const defaultState = {
  editorState: JSON.stringify({ recipe, equipment }, null, 4)
}

const updateEditor = (state = defaultState, { payload, type }) => {
  if (type === 'UPDATE_EDITOR_STATE') {
    return {
      ...state,
      editorState: payload
    }
  }
  return state
}

export const persistedState = localStorage.getItem('brewCalcState')
  ? JSON.parse(localStorage.getItem('brewCalcState'))
  : defaultState

export default updateEditor
