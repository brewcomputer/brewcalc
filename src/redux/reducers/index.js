import { combineReducers } from 'redux'
import { routerReducer } from 'react-router-redux'

export default combineReducers({
  router: routerReducer,
})
