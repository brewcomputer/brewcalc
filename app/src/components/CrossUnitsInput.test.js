import React from 'react'
import renderer from 'react-test-renderer'

import { createStore } from 'redux'
import updateEditor from '../redux/reducers/updateEditor'
import { Provider } from 'react-redux'

import CrossUnitsInput from './CrossUnitsInput'

const store = createStore(
  updateEditor
)

it('renders without crashing', () => {
  const tree = renderer
    .create(
    <Provider store={store}>
      <CrossUnitsInput description="description" name="Boil Size" value={20} unit="L" />
    </Provider>
    )
    .toJSON()
  expect(tree).toMatchSnapshot()
})

it('renders without crashing on US units', () => {


  store.dispatch({
    type: 'UPDATE_UNITS',
    payload: 'english'
  })

  const tree = renderer
    .create(<Provider store={store}>
      <CrossUnitsInput description="description" name="Boil Size" value={20} unit="L" />
    </Provider>
    )
    .toJSON()
  expect(tree).toMatchSnapshot()
})