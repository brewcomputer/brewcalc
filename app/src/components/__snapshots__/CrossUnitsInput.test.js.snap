// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders without crashing 1`] = `
<div
  title="description"
>
  <b>
    Boil Size
     
  </b>
  20
   
  L
</div>
`;

exports[`renders without crashing on US units 1`] = `
<div
  title="description"
>
  <b>
    Boil Size
     
  </b>
  5.28
   
  gal
</div>
`;
